File: kakao-ios-sdk-1/samples/objc/KakaoSample/KakaoSample/TalkProfileTableViewCell.m
/**
 * Copyright 2015-2016 Kakao Corp.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#import "TalkProfileTableViewCell.h"
#import "UIImageView+WebCache.h"

@interface TalkProfileTableViewCell ()

@property (weak, nonatomic) IBOutlet UILabel *nickName;
@property (weak, nonatomic) IBOutlet UILabel *countryISO;

@end

@implementation TalkProfileTableViewCell

- (void)awakeFromNib {
    [super awakeFromNib];
}

- (void)setTalkProfile:(KOTalkProfile *)profile {
    if (profile == nil) {
        return;
    }
    
    if (profile.thumbnailURL) {
        [self.thumbnail sd_setImageWithURL:[NSURL URLWithString:profile.thumbnailURL]
                          placeholderImage:[UIImage imageNamed:@"PlaceHolder"]];
    } else {
        self.thumbnail.image = [UIImage imageNamed:@"PlaceHolder"];
    }
    
    self.nickName.text = profile.nickName;
    self.countryISO.text = profile.countryISO;
}

@end

File: kakao-ios-sdk-1/samples/swift/KakaoSample/KakaoSample/TalkProfileTableViewCell.swift
/**
* Copyright 2015 Kakao Corp.
*
* Redistribution and modification in source or binary forms are not permitted without specific prior written permission.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*    http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

import UIKit

class TalkProfileTableViewCell: UITableViewCell {
    @IBOutlet weak var thumbnail: UIImageView!
    @IBOutlet weak var nickName: UILabel!
    @IBOutlet weak var countryISO: UILabel!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }
    
    func setTalkProfile(_ profile: KOTalkProfile!) {
        self.thumbnail.image = UIImage(named: "PlaceHolder")
        
        if profile == nil {
            return
        }
        
        if profile.thumbnailURL != nil && profile.thumbnailURL.isEmpty == false {
            if let thumbnailURL = URL(string: profile.thumbnailURL!) {
                self.thumbnail.setImage(withUrl: thumbnailURL)
            }
        }
        
        if profile.nickName != nil {
            self.nickName.text = profile.nickName
        }
        
        if profile.countryISO != nil {
            self.countryISO.text = profile.countryISO
        }
    }    
}
